### DISCUSSION TITLE
How to create or connect a tag using prisma?

### DISCUSSION THREAD

**Discussion by gustavogmfarias** [prisma/prisma#17014] [Q&A]
Hey guys, 'im having a lot of problemas with that and the solutions i find on internet didn't work for me, or i didn't understand.

I'm brazilian first of all, and althoung i can read in english, there are words or tems i don't understand.

But, let's get into the problem.

This is my scheme:

`model Article {
    id                     String                   @id @default(uuid())
    title                  String
    subTitle               String
    content                String
    images                 ArticleImages[]
    mainImage              String?
    publishedBy            User                     @relation("Publisher", fields: [publishedByUserId], references: [id], onDelete: SetDefault)
    editedBy               User?                    @relation("Editor", fields: [editedByUserId], references: [id], onDelete: SetNull)
    publishedByUserId      String
    editedByUserId         String?
    publishedDate          DateTime                 @default(now())
    updatedDate            DateTime?                @default(now())
    isHighlight            Boolean                  @default(true)
    url                    String
    AuthorsOnArticles      AuthorsOnArticles[]
    TagsOnArticles         TagsOnArticles[]
    CoursesOnArticles      CoursesOnArticles[]
    ClassOnArticles        ClassOnArticles[]
    CategoryOnArticles     CategoryOnArticles[]
    TextualGenreOnArticles TextualGenreOnArticles[]

    @@map("articles")
}`

`model TagsOnArticles {
    Article   Article @relation(fields: [articleId], references: [id], onDelete: Cascade)
    articleId String
    tag       Tag     @relation(fields: [tagId], references: [id])
    tagId     String

    @@id([tagId, articleId])
}`

And what i want to do is adding  a new article with tags.

If the tag exists, only connect. If the tags doesn't exsit, it is created and connected to the article.
I did a bad code before to make it working, but now that form is returning a lot of problems, so i need too much to make it woks.

Please, it is my college paper.

The repository to create is that:
`    async create({
        title,
        subTitle,
        content,
        publishedByUserId,
        isHighlight,
        url,
    }: ICreateArticleDTO): Promise<ArticleWithRelations> {
        const article = await prisma.article.create({
            data: {
                title,
                subTitle,
                content,
                publishedByUserId,
                isHighlight,
                url,
            },
        });`
        
        Please, help me, with this solution a can resolve the  other ones that i have, like category, etc.
        
My git: https://github.com/gustavogmfarias/iffolha-js-backend/blob/6647f6c1ef71ba79196d9cb04123610b3584fd5c/src/modules/articles/repositories/infra/ArticleRepository.ts#L31

**Comment by gustavogmfarias** [comment]
Ah, this is the body.params to create:

`{
						"title": "gg gust3vo",
            "subTitle": "a cruz de malta é meu pendão",
            "content": "tu tens o nome do heroico portugues",
            "isHighlight": false, 
						"authors": [], 
	"tags": ["joão", "tiago"]

}`

![image](https://user-images.githubusercontent.com/53410368/209453102-2acbb3a8-d3ca-4ed7-a9e4-784ad48b8150.png)

**Reply by nurul3101** [reply]
Hello @gustavogmfarias 👋 
What exact error are you getting? Can you share the error message?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
